wordsFrequency/main.cpp

```diff
@@ -0,0 +1,129 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <utility>
+#include <vector>
+#include <cctype>
+#include <list>
+#include <algorithm>
+#include <limits>
+#include <optional>
+#include <set>
+#include <cassert>
+
+void addToFirstTen(std::vector<std::pair<std::string, int>>& words, const std::string& word, int frequency) {
+    bool inserted = false;
+
+    for(ulong i = 0; i < words.size(); i++) {
+        if(frequency > words[i].second)
+        {
+            auto it = words.begin() + i;
+            words.insert(it, std::pair(word, frequency));
+            inserted = true;
+            break;
+        }
+    }
+
+    if(!inserted)
+        words.push_back(std::pair(word, frequency));
+
+    if(words.size() > 10)
+        words.pop_back();
+}
+
+void collectMostFrequent(const std::string& fileName, std::vector<std::pair<std::string, int>>& firstTen) {
```

Comment:
In C++11 puoi ritornare un vector senza incorrere in penalità di performance.
```suggestion
std::vector<std::pair<std::string, int>> collectMostFrequent(const std::string& fileName) {
```